src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -82,78 +87,73 @@ public void update() {
+                .child(Flow.column()
+                        .top(28)
+                        .left(10).right(10)
+                        .coverChildrenHeight()
+                        .child(createMinMaxRow("cover.advanced_energy_detector.min",
+                                this::getMinValue, this::setMinValue,
+                                this::getPostFix, this::updateWidget))
+                        .child(createMinMaxRow("cover.advanced_energy_detector.max",
+                                this::getMaxValue, this::setMaxValue,
+                                this::getPostFix, this::updateWidget))
+                        .child(Flow.row()
+                                .widthRel(1f)
+                                .coverChildrenHeight()
+                                .marginBottom(5)
+                                .child(IKey.lang("cover.advanced_energy_detector.modes_label").asWidget()
+                                        .size(72, 18))
+                                .child(new ToggleButton()
+                                        .right(0)
+                                        .size(72, 18)
+                                        .addTooltipLine(IKey.lang("cover.advanced_energy_detector.modes_tooltip"))
+                                        .value(new BooleanSyncValue(this::isUsePercent, this::setUsePercent))
+                                        .overlay(new DynamicDrawable(() -> IKey
+                                                .lang("cover.advanced_energy_detector.mode_" +
+                                                        (isUsePercent() ? "percent" : "eu"))
+                                                .style(TextFormatting.WHITE)))))
+                        .child(Flow.row()
+                                .widthRel(1f)
+                                .coverChildrenHeight()
+                                .child(IKey.lang("cover.generic.advanced_detector.invert_label").asWidget()
+                                        .size(72, 18))
+                                .child(new ToggleButton()
+                                        .right(0)
+                                        .size(72, 18)
+                                        .addTooltipLine(IKey.lang("cover.advanced_energy_detector.invert_tooltip"))
+                                        .value(new BooleanSyncValue(this::isInverted, this::setInverted))
+                                        .overlay(new DynamicDrawable(() -> IKey
+                                                .lang("cover.advanced_energy_detector." +
+                                                        (isInverted() ? "inverted" : "normal"))
+                                                .style(TextFormatting.WHITE))))))
+                .bindPlayerInventory();
+    }
+
+    private void updateWidget(GTTextFieldWidget w) {
+        w.setMaxLength(getLength());
+        w.setNumbers(0, isUsePercent() ? 100 : Integer.MAX_VALUE);
```

Comment:
Energy is the one thing that I can see being a long in these covers, so why cap at max int?

Reply:
our text field widget was missing a method to for setting min and max with long values